OSDOCS-4928: Docs for cert-manager operator GA (#56754)
openshift/openshift-docs: security/cert_manager_operator/cert-manager-operator-release-notes.adoc

These release notes track the development of {cert-manager-operator}.

For more information, see xref:../../security/cert_manager_operator/index.adoc#cert-manager-operator-about[About the {cert-manager-operator}].

[id="cert-manager-operator-release-notes-1.10.2"]
== Release notes for {cert-manager-operator} 1.10.2

// TODO: Confirm date closer to GA
Issued: 2023-03-23

The following advisory is available for the {cert-manager-operator} 1.10.2:

// TODO: Update
* link:https://access.redhat.com/errata/RHEA-2023:TODO[RHEA-2023:TODO]
https://errata.devel.redhat.com/advisory/109104

For more information, see the link:https://cert-manager.io/docs/release-notes/release-notes-1.10[cert-manager project release notes for v1.10].

[id="cert-manager-operator-1.10.2-new-features-and-enhancements"]
=== New features and enhancements

// * This is the GA release of the {cert-manager-operator}.
* TODO

[id="cert-manager-operator-1.10.2-bug-fixes"]
=== Bug fixes

> * TODO

Author: @thejasn @swghosh Do we have a list of bug fixes compared to the old TP version that we can list here?

Reviewer: I've cc'ed you on slack.

Author: Thanks @thejasn. Can you update each of these with some basic release notes text?

We're usually looking for it in the Cause/Consequence/Fix/Result (CCFR) format and we usually use the "release notes text" field, but I don't have permissions to the CM project to see if that field exists.

Again, doesn't have to be anything spectacular since I will take a pass at updating the actual wording. I just need to know a good summary of what the bug was vs now that it's resolved.

[id="cert-manager-operator-1.10.2-known-issues"]
=== Known issues

// TODO: Check if this is still appropriate to list
* Using `Route` objects is not fully supported. Currently, {cert-manager-operator} integrates with `Route` objects by creating `Ingress` objects through the Ingress Controller. (link:https://issues.redhat.com/projects/CM/issues/CM-16[*CM-16*])

> * TODO: Any other known issues to list?

Author: @thejasn @swghosh Is the above known issue still appropriate to list?

Are there any other known issues that we should list here?

Reviewer: Yes, routes are still not supported. For now this is it.

// TODO: Any other features to list or things to call out as supported? proxy support?

[id="cert-manager-operator-1.10.2-bug-fixes"]
=== Bug fixes

* Previously, the `unsupportedConfigOverrides` field replaced user-provided arguments instead of appending them. Now, the `unsupportedConfigOverrides` field properly appends user-provided arguments. (*link:https://issues.redhat.com/browse/CM-23[CM-23]*)
+
[WARNING]
====
Using the `unsupportedConfigOverrides` section to modify the configuration of an Operator is unsupported and might block cluster upgrades.
====

* Previously, the {cert-manager-operator} did not inject the necessary cluster wide egress proxy configuration. With this release, the {cert-manager-operator} now properly supports injecting custom CA certificates and detects cluster wide egress proxy configuration. (*link:https://issues.redhat.com/browse/CM-37[CM-37]*)

* Previously, the {cert-manager-operator} was installed as a cluster Operator. With this release, the {cert-manager-operator} is now properly installed as an OLM Operator. (*link:https://issues.redhat.com/browse/CM-35[CM-35]*)

[id="cert-manager-operator-1.10.2-known-issues"]
=== Known issues

> * Using `Route` objects is not fully supported. Currently, {cert-manager-operator} integrates with `Route` objects by creating `Ingress` objects through the Ingress Controller. (link:https://issues.redhat.com/projects/CM/issues/CM-16[*CM-16*])

Reviewer 2: https://issues.redhat.com/projects/CM/issues/CM-16 will display other issues on the left. https://issues.redhat.com/browse/CM-16 is consistent with previous paragraphs' links.
```suggestion
* Using `Route` objects is not fully supported. Currently, {cert-manager-operator} integrates with `Route` objects by creating `Ingress` objects through the Ingress Controller. (link:https://issues.redhat.com/browse/CM-16[*CM-16*])
```

Issued: 2023-03-23

The following advisory is available for the {cert-manager-operator} 1.10.2:

* link:https://access.redhat.com/errata/RHEA-2023:1238[RHEA-2023:1238]

For more information, see the link:https://cert-manager.io/docs/release-notes/release-notes-1.10[cert-manager project release notes for v1.10].

[IMPORTANT]
====
If you used the Technology Preview version of the {cert-manager-operator}, you must uninstall and move all related resources for the Technology Preview version before installing this version of the {cert-manager-operator}.

For more information, see xref:../../security/cert_manager_operator/cert-manager-operator-uninstall.adoc#cert-manager-operator-uninstall[Uninstalling the {cert-manager-operator}].
====

[id="cert-manager-operator-1.10.2-new-features-and-enhancements"]
=== New features and enhancements

This is the general availability (GA) release of the {cert-manager-operator}.

* The following issuer types are supported:
** ACME
** CA
** Self-signed

* The following ACME challenge types are supported:
** DNS-01
** HTTP-01

* The following DNS-01 providers for ACME issuers are supported:
** Amazon Route 53
** Azure DNS
** Google Cloud DNS

> // TODO: Any other features to list or things to call out as supported? proxy support?

Author: @thejasn @swghosh I added a few things that we support to this list in the release notes. Is there anything else worthwhile that you think we should call out here?

Reviewer: Both supporting cluster wide egress proxy and injecting custom CA are called out in bug fixes, so kind of becomes redundant.

Reviewer: Both 
```
** Azure DNS
** Google Cloud DNS
```
Are supported using specific methods only, maybe we can add a footnote for this.

Author: @thejasn would it be better to call these features and move them up here? Or are they more accurate labeled as bug fixes?

As for your second comment, I do have the details listed in the known issues, but then also put that info in the main supported DNS providers section too. Let me know if that is sufficient: https://56754--docspreview.netlify.app/openshift-enterprise/latest/security/cert_manager_operator/cert-manager-operator-issuer-acme.html#cert-manager-acme-dns-providers_cert-manager-operator-issuer-acme

> * The {cert-manager-operator} now supports injecting custom CA certificates for the cluster-wide egress proxy configuration.

Reviewer 2: @bergerhoffer @thejasn need it mention `supports injecting egress proxy env vars automatically` as well?

Reviewer: This gets documented as across all operators https://docs.openshift.com/container-platform/4.12/operators/admin/olm-configuring-proxy-support.html, probably we can call out that this is supported now.

Author: I added "and egress proxy environment variables" - let me know if that sounds okay.

Reviewer: ["and propagates cluster wide egress proxy environment variables" ](https://github.com/openshift/openshift-docs/pull/56754/files#r1142397268)

> * The {cert-manager-operator} now supports injecting custom CA certificates and egress proxy environment variables for the cluster-wide egress proxy configuration.

Reviewer: ```suggestion
* The {cert-manager-operator} now supports injecting custom CA certificates and propagates cluster-wide egress proxy environment variables to cert-manager.
```

Author: Updated!

[id="cert-manager-operator-1.10.2-bug-fixes"]
=== Bug fixes

* Previously, the `unsupportedConfigOverrides` field replaced user-provided arguments instead of appending them. Now, the `unsupportedConfigOverrides` field properly appends user-provided arguments. (*link:https://issues.redhat.com/browse/CM-23[CM-23]*)
+
[WARNING]
====
Using the `unsupportedConfigOverrides` section to modify the configuration of an Operator is unsupported and might block cluster upgrades.
====

* Previously, the {cert-manager-operator} was installed as a cluster Operator. With this release, the {cert-manager-operator} is now properly installed as an OLM Operator. (*link:https://issues.redhat.com/browse/CM-35[CM-35]*)

[id="cert-manager-operator-1.10.2-known-issues"]
=== Known issues

> * Using `Route` objects is not fully supported. Currently, {cert-manager-operator} integrates with `Route` objects by creating `Ingress` objects through the Ingress Controller. (link:https://issues.redhat.com/browse/CM-16[*CM-16*])

Reviewer: ```suggestion
* Using `Route` objects is not fully supported. Currently to use {cert-manager-operator} with `Routes`, users need to create `Ingress` objects which will be translated to a Route by the Ingress-to-Route Controller. (link:https://issues.redhat.com/browse/CM-16[*CM-16*])
```

Author: Updated (tweaked suggestion slightly)

* {cert-manager-operator} does not support using Azure Active Directory (Azure AD) pod identities to assign a managed identity to a pod. As a workaround, you can use a service principal to assign a managed identity. (link:https://issues.redhat.com/browse/OCPBUGS-8665[*OCPBUGS-8665*])

> * {cert-manager-operator} does not support using Google Kubernetes Engine (GKE) Workload Identity. (link:https://issues.redhat.com/browse/OCPBUGS-9998[*OCPBUGS-9998*])

Reviewer 2: Same as https://github.com/openshift/openshift-docs/pull/56754/files#r1141756631

* {cert-manager-operator} does not support using Google Kubernetes Engine (GKE) Workload Identity (link:https://issues.redhat.com/browse/OCPBUGS-9998[*OCPBUGS-9998*])

> 

Reviewer 2: @bergerhoffer since we have security/cert_manager_operator/cert-manager-operator-uninstall.adoc, but there is [OCPBUGS-9960](https://issues.redhat.com/browse/OCPBUGS-9960) about uninstall, and the bug can be easily hit by users in the user experience perspective, I suggest list this bug as known issue.

Reviewer 2: @bergerhoffer , another bug about dns01, [OCPBUGS-8069](https://issues.redhat.com/browse/OCPBUGS-8069), which had Slack discussion link with PM/Dev, may need be listed as known issue

Reviewer: The bug on the private DNS hosted zone need not be called under "Known issues" since this isn't really an issue it  is more of a invalid use case for DNS-01 type challenges (as mentioned by upstream as well). This limitation can be provided as a note in the DNS-01 section though.

Author: Thank you both.

When I tested uninstalling, I didn't see that checkbox option - does it only show in certain circumstances?

Either way, I noted it as a known issue. I've also added a note in the DNS-01 section saying that private DNS zones aren't supported.
